contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -113,14 +113,17 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
 
         // Contract creations are signalled by sending a transaction to the zero address.
         if (decodedTx.to == address(0)) {
-            address created = Lib_SafeExecutionManagerWrapper.safeCREATE(
+            (address created, bytes memory revertData) = Lib_SafeExecutionManagerWrapper.safeCREATE(
                 decodedTx.gasLimit,
                 decodedTx.data
             );
 
-            // EVM doesn't tell us whether a contract creation failed, even if it reverted during
-            // initialization. Always return `true` for our success value here.
-            return (true, abi.encode(created));
+            // Return true if the contract creation succeeded, false w/ revertData otherwise.
+            if (created != address(0)) {
+                return (true, abi.encode(created));
```

Comment:
We don't technically need to return this address here, can leave it if we want.

Reply:
Yeah, I lean towards leaving it.  You can see how it's actually done even differently in `eth_call`, see the code lookup in the `simulateMessage` case.